tests/main/i18n/task.yaml

```diff
@@ -26,3 +26,14 @@ execute: |
         fi
     done
     not cat bad
+
+    echo "Lint translated commands"
+    # Only attempt to install the langpacks in Ubuntu
+    if apt install language-pack-en; then
+      # it's fine to use ls here since the filenames are very simple
+      # shellcheck disable=SC2011
+      ls /usr/share/locale/ | xargs -Iloc apt install language-pack-loc -y
```

Comment:
This is a fine approximation but I feel it's a bit suboptimal because:
a) there will be a lot of apt install runs that will return an error but xargs will not catch it, when running this on my system with `ls /usr/share/locale/ | xargs -Iloc sh -c "apt install -s language-pack-loc || exit 255"` (to stop on failed commands) then it already fails on the first entry of the dir
b) we cannot cleanup the packages because we don't know what we installed.

Maybe we can do something like:
```
diff --git a/tests/main/i18n/task.yaml b/tests/main/i18n/task.yaml
index 4106f04777..f52860d635 100644
--- a/tests/main/i18n/task.yaml
+++ b/tests/main/i18n/task.yaml
@@ -3,6 +3,9 @@ summary: Test that i18n works
 # no i18n yet in debian-sid
 systems: [-debian-sid-*]
 
+restore: |
+    apt remove -y 'language-pack-[a-z]+$'
+
 execute: |
     # The snapd deb from the archive does not contain .mo files, those
     # are stripped out by the langpack buildd stuff and put into the
@@ -29,10 +32,8 @@ execute: |
 
     echo "Lint translated commands"
     # Only attempt to install the langpacks in Ubuntu
-    if apt install language-pack-en; then
-      # it's fine to use ls here since the filenames are very simple
-      # shellcheck disable=SC2011
-      ls /usr/share/locale/ | xargs -Iloc apt install language-pack-loc -y
+    if os.query is_ubuntu; then
+        apt install -y 'language-pack-[a-z]+$'
     fi
 
     # shellcheck disable=SC2002
```
Fwiw, when I run this locally in spread I got:
```
2021/09/15 09:36:40.652406 main.go:588: PANIC description of prepare-image's "<target-dir>" is lowercase in locale "pt_BR ISO-8859-1": "o directório de destino"
panic: description of prepare-image's "<target-dir>" is lowercase in locale "pt_BR ISO-8859-1": "o directório de destino" [recovered]
	panic: description of prepare-image's "<target-dir>" is lowercase in locale "pt_BR ISO-8859-1": "o directório de destino"
```
but I am not sure if I maybe just did something wrong here.

Reply:
Heh, I just looked at the panic and it seems like we panic if snapdenv.Testing() is set and it's already enough if we build with testkeys for this to be true which I think we do in our spread tests. So doing the end-to-end test for this (that it really does not panic) needs a bit more tweaking :)

Reply:
Thanks for the suggestion, this is much better than my hack =)

> Heh, I just looked at the panic and it seems like we panic if snapdenv.Testing() is set and it's already enough if we build with testkeys for this to be true which I think we do in our spread tests. So doing the end-to-end test for this (that it really does not panic) needs a bit more tweaking :)

Sorry, I didn't get your point. The panic is expected because of the issue with the pt_br translation. Or did you mean that the SNAPD_DEBUG=1 can be removed since snapd is already built with the testkeys build flag?